src/library/handlers/traceHandler.ts

```diff
@@ -145,42 +155,51 @@ export class TraceHandler {
             }
         }
         const ctx = context.active();
-        let attributes: SpanAttributes = {
+        let attributes: Attributes = {
             ...telemetry.properties,
         };
-        if (telemetry.dependencyTypeName.toLowerCase().indexOf("http") > -1) {
-            attributes[SemanticAttributes.HTTP_METHOD] = "http";
-            try {
-                let url = new URL(telemetry.data);
-                attributes[SemanticAttributes.HTTP_METHOD] = url.protocol;
-            } catch (error) {
-                // Ignore error
+        if (telemetry.dependencyTypeName) {
+            if (telemetry.dependencyTypeName.toLowerCase().indexOf("http") > -1) {
```

Comment:
This is kind of strange to me. Are we mapping from `DependencyTelemetry` (AppInsights) to OpenTelemetry spans and then back to `DependentTelemetry` when we export using the Azure monitor exporter?

Reply:
That is correct, that would be the only way to allow a manual API right now